dashboard/signals_test.py

```diff
@@ -2,91 +2,205 @@
+            cls.course_run = CourseRunFactory.create(program=cls.program)
+            cls.separate_course_run = CourseRunFactory.create()
+
+    def test_create_linked(self):
+        """
+        The database is reindexed when a CachedCertificate is created and linked to a ProgramEnrollment
+        """
+        with patch('search.tasks._index_program_enrolled_users', autospec=True) as mocked:
+            CachedCertificateFactory.create(user=self.user, course_run=self.course_run)
+        mocked.assert_called_once_with([self.program_enrollment])
+
+    def test_create_unlinked(self):
+        """
+        The database is not reindexed when a CachedCertificate is created but not linked to a ProgramEnrollment
+        """
+        with patch('search.tasks._index_program_enrolled_users', autospec=True) as mocked:
+            CachedCertificateFactory.create(user=self.user, course_run=self.separate_course_run)
+        assert not mocked.called
+
+    def test_update_linked(self):
+        """
+        The database is reindexed when a CachedCertificate is updated and linked to a ProgramEnrollment
+        """
+        with mute_signals(post_save):
+            certificate = CachedCertificateFactory.create(user=self.user, course_run=self.course_run)
+        with patch('search.tasks._index_program_enrolled_users', autospec=True) as mocked:
+            certificate.save()
+        mocked.assert_called_once_with([self.program_enrollment])
+
+    def test_update_unlinked(self):
+        """
+        The database is not reindexed when a CachedCertificate is updated but not linked to a ProgramEnrollment
+        """
+        with mute_signals(post_save):
+            certificate = CachedCertificateFactory.create(user=self.user, course_run=self.separate_course_run)
+        with patch('search.tasks._index_program_enrolled_users', autospec=True) as mocked:
+            certificate.save()
+        assert not mocked.called
+
+    def test_delete_linked(self):
+        """
+        The database is reindexed when a CachedCertificate is deleted when linked to a ProgramEnrollment
+        """
+        with mute_signals(post_save):
+            certificate = CachedCertificateFactory.create(user=self.user, course_run=self.course_run)
+        with patch('search.tasks._index_program_enrolled_users', autospec=True) as mocked:
+            certificate.delete()
+        mocked.assert_called_once_with([self.program_enrollment])
+
+    def test_delete_unlinked(self):
+        """
+        The database is not reindexed when a CachedCertificate is deleted when not linked to a ProgramEnrollment
+        """
+        with mute_signals(post_save):
+            certificate = CachedCertificateFactory.create(user=self.user, course_run=self.separate_course_run)
+        with patch('search.tasks._index_program_enrolled_users', autospec=True) as mocked:
+            certificate.delete()
+        assert not mocked.called
```

Comment:
this class is another 65 lines of code that we've copied/pasted because we don't parameterize our tests. i won't hold up this PR for it, but it's worth pointing out